### PR TITLE
Fix broken BitString

### PR DESCRIPTION
**Pull request by matejcik**
This PR adds a couple testcases for BitString that were broken for a very long time ... and fixes them :)

Namely:
* it was impossible to create a correct bitstring with multiple namedValues set, because the algorithm for setting bits was flawed
* it was not possible to use the syntax `group1 = Rights(('group-read', 'group-write'))`, explicitly mentioned in documentation, because the values would be treated as truthy bits instead of referring to namedValues

In addition, I'm adding pretty-printing of named bits.

### CODE REVIEW

File: pyasn1/type/univ.py [hunk 1]
```diff
@@ -676,6 +666,19 @@ def prettyIn(self, value):
                 return self.fromBinaryString(value, internalFormat=True)
 
         elif isinstance(value, (tuple, list)):
+            if self.namedValues and all(isinstance(v, str) for v in value):
+                bits = [self.namedValues[name] for name in value]
```
Comment (droideck) [inline pyasn1/type/univ.py]: I think this part around `[self.namedValues[name] for name in names]` was useful, or do I miss something?

```python
except KeyError:
    raise error.PyAsn1Error('unknown bit name(s) in %r' % (names,))
```

**Review by droideck — COMMENTED** [review]
Otherwise - looks great!

**Review by droideck — COMMENTED** [review]
Hey @matejcik, thanks again for this fix!
I found one small edge case while testing: passing an empty list/tuple when `namedValues` is defined causes a crash:

```python
b = BitString(namedValues=NamedValues(('Active', 0), ('Urgent', 1)))
b.clone([])  # ValueError
```

This happens because `max(bits)` is called on an empty list at line 674.
Would you mind adding a quick guard for this? Something like:

```python
  if not bits:
      return SizedInteger(0).setBitLength(0)
```

And maybe a small test case:

```python
    assert self.b.clone([]) == ()
    assert self.b.clone(()) == ()
```

I know it's a bit of a corner case, but it would be nice to handle it gracefully.

**Comment by matejcik** [comment]
done in 6ef189ccde5cad435174dcb84bc007c6a000b0c7, i also rebased on top of current main
